src/parsers/archaea_parsers.py

```diff
@@ -0,0 +1,564 @@
+import uuid
+import xml.etree.ElementTree as ET
+from datetime import date
+import json
+import requests
+import gzip
+from pyspark.sql import SparkSession
+from delta import configure_spark_with_delta_pip
+from pyspark.sql.types import StringType
+import os
+import shutil
+from pyspark.sql.types import (ArrayType, StructType, StructField)
+
+
+NS = {"u": "https://uniprot.org/uniprot"}
+
+def generate_unique_id():
+    """
+    Generate a unique identifier for the CDM.
+    """
+    return f"CDM:{uuid.uuid4()}"
+
+
+def build_datasource_record():
+    """
+    Construct provenance information for the data source.
+    Add in DataSource Table with provenance information for Uniprot download.
+    """
+    return {
+        "name": "UniProt archaea",
+        "source": "UniProt",
+        "url": "https://uniprot.org/downloads/path/to/file",
+        "accessed": date.today().strftime("%Y-%m-%d"),
+        "version": 115,
```

Comment:
Some of this should come from the data source itself or from the URL that you use to access it -- `name`, `url`, and `version`.

For the `accessed` field, use a POSIX timestamp - much easier than faffing around with date formatting!

Reply:
Here's some code for it:

https://github.com/kbase/cdm-task-service/blob/main/cdmtaskservice/timestamp.py